.github/workflows/testing.yml

```diff
@@ -0,0 +1,73 @@
+name: Full Tests
+on:
+  schedule:
+    - cron: "0 0 * * *"  # daily
+  pull_request:
+
+jobs:
+  run:
+    name: Full tests on ${{ matrix.os }} with Python ${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+        os: [ubuntu-latest, windows-latest] # macos-latest; problems with git-annex and all tests require data here (no minimal/internals)
+    steps:
+      - uses: s-weigand/setup-conda@v1
+      - uses: actions/checkout@v2
+      - run: git fetch --prune --unshallow --tags
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - if: ${{ matrix.os == 'ubuntu-latest' }}
+        name: Get datalad - Linux
+        run: conda install -c conda-forge datalad==0.16.1
+      - if: ${{ matrix.os == 'windows-latest' }}
+        name: Get git-annex - Windows
+        uses: crazy-max/ghaction-chocolatey@v1.6.0
+        with:
+          args: install git-annex --ignore-checksums
+      - if: ${{ matrix.os == 'windows-latest' }}
+        name: Get datalad - Windows and Mac
+        run: pip install datalad==0.15.6
+
+      - name: Global Setup
+        run: |
+          pip install -U pip
+          pip install pytest-xdist
+          git config --global user.email "CI@example.com"
+          git config --global user.name "CI Almighty"
+      - name: Test minimal installation
+        run: pip install .
+      - name: Test full installation
+        run: pip install .[full]
+      - name: Install testing requirements (-e needed for codecov report)
+        run: pip install -e .[testing]
+
+      - name: Get ophys_testing_data current head hash
+        id: ophys
+        run: echo "::set-output name=HASH_OPHYS_DATASET::$(git ls-remote https://gin.g-node.org/CatalystNeuro/ophys_testing_data.git HEAD | cut -f1)"
+      - name: Cache ophys dataset - ${{ steps.ophys.outputs.HASH_OPHYS_DATASET }}
+        uses: actions/cache@v2
+        id: cache-ophys-datasets
+        with:
+          path: ./ophys_testing_data
+          key: ophys-datasets-051822-${{ matrix.os }}-${{ steps.ophys.outputs.HASH_OPHYS_DATASET }}
```

Comment:
You know, when I saw the date here for the first time I could not tell it was a date because of the format : )

Reply:
lol that's just my personal way of associating a manual hash to it - it doesn't *have* to be a date per se. In fact I used to just use a running integer count, but I figured a date would give more contextual detail